src/js/ripple/amount.js

```diff
@@ -176,13 +189,7 @@ Amount.prototype.ratio_human = function(denominator, opts) {
   opts = extend({ }, opts);
 
   var numerator = this;
-
-  if (typeof denominator === 'number' && parseInt(denominator, 10) === denominator) {
-    // Special handling of integer arguments
-    denominator = Amount.from_json(String(denominator) + '.0');
-  } else {
-    denominator = Amount.from_json(denominator);
-  }
+  denominator = Amount.from_json(noIntegers(denominator));
```

Comment:
It does indeed seem a *_human, force XRP scale for native amounts hack as you said.